### ISSUE TITLE
Please make the name optional on the signUp on both the client and server 

### ISSUE THREAD

**Issue by firstaxel** [better-auth/better-auth#424]

**Comment by firstaxel** [comment]
not all signUp as i have seen doesnt ask for the user name

**Comment by itsderinger** [comment]
@firstaxel, how did you resolve this? facing the same issue.

**Comment by firstaxel** [comment]
you can bypass this error by 
`
name: ""
`
the error is going to go

**Comment by itsderinger** [comment]
> you can bypass this error by `name: ""` the error is going to go

I figured, but that's really counter-intuitive, hence making the field optional in the config is still the better approach, imo.

**Comment by itsderinger** [comment]
@Bekacru What's your take? Would be cool to have it before v1?

**Comment by chris-verclytte** [comment]
Same here, you might easily have a user flow that just ask for the email and password and provide further details afterwards to limit the UX friction.

Maybe something like : 
```js 
export const auth = betterAuth({
   user: {
    fields: {
      name: { required: false } // provide options instead of a simple string for mapping
    },
  },
})

// or 

export const auth = betterAuth({
   user: {
    additionalFields: {
      name: { required: false } // override default behavior when set in additionalFields
    },
  },
})
```

But it might even be preferable to get rid of the name in `better-auth` as you might want to split name with first_name / last_name in your DB and having name create an unnecessary column in the db.

**Comment by firstaxel** [comment]
Thanks this is true please fix it

**Comment by ping-maxwell** [comment]
Totally agree with this, in my case, I have a `first_name` and `last_name`. I don't need the `name` field at all.

**Comment by zispidd** [comment]
@Bekacru any updates on this and "image" field?

**Comment by ping-maxwell** [comment]
There are future plans to make the fields dynamic.
Not sure when this will be a thing, but it's definitely on @Bekacru radar and apart of the plan.

**Comment by zispidd** [comment]
still relevant

**Comment by Makakusnik** [comment]
+1 
Plugins should do bare minimum but they should be opened to any adjustments.. for example emailAndPassword plugin should be only responsible for emailAndPassword i really really dont understand need of name being in there too, and image aswell but that's optional at least..

**Comment by firstaxel** [comment]
Still relevant it has been changed

**Comment by firstaxel** [comment]
I mean it hasn't been changed

**Comment by tomleidy** [comment]
I’m really interested in this change, too. Requiring a name field when it’s "emailAndPassword" seems very strange. Looking forward to this getting implemented.

**Comment by JPustkuchen** [comment]
I agree that for man SaaS cases you don't want a "Name", but just the email address (and maybe dedicated address fields). Name should be removed here or at least be made optional / nullable and left out in the object keys.

**Comment by DNoel26** [comment]
Has this been changed? Before I bypassed the issue with setting `name: ""` but now I get the error: `[body.name] Too small: expected string to have >=1 characters`

**Comment by firstaxel** [comment]
It has not been fixed

**Comment by DNoel26** [comment]
> It has not been fixed
> […](#)

I know, that's why I passed an empty string. But as of recently I need to pass at least 1 character or I'd get the error I mentioned.

**Comment by JE4GLE** [comment]
@Bekacru +1 Since the empty string isn't working any more, this issue is getting more and more annoying

**Comment by JE4GLE** [comment]
I've created this temporary fix.

```typescript
export const auth = betterAuth({
	databaseHooks: {
		user: {
			create: {
				before: async (user, ctx) => {
					if (ctx?.path === '/sign-up/email') {
						// Temporary fix for better-auth issue #424
						if (user.name === user.email) {
							return {
								data: {
									...user,
									name: undefined
								}
							};
						}
					}
				}
			}
		}
	}
});
```

Just set the name the same as the email:
```typescript
authClient.signUp.email(
  {
  	name: username,
  	email: username,
  	password: password,
  }
);
```

**Comment by xsevy** [comment]
would love to see this in future release.

**Comment by switz** [comment]
This is a silly requirement, `name` is not a functional requirement in all applications.

**Comment by Jlh77** [comment]
In the meantime, I believe you could do this:

 In your `auth.ts` config:

```
...
user: {
  additionalFields: {
    name: {
      type: "string",
      required: false,
      input: false,
    },
  },
},
...
```

You can just overwrite the name field and make it not needed as an input.

And if you don't want it at all, you could remove it from your schema manually (depending on what your adapter is)

**Comment by Jlh77** [comment]
The fact that creating a user with `name: ""` is no longer accepted seems weird considering that you can still set `name` to blank when using `updateUser`.

Either it should be one behaviour or the other.

**Comment by bytaesu** [comment]
We’re discussing including this in the next release 🙂

**Comment by bytaesu** [comment]
> [!NOTE]
> We'll either find a way to do this without a breaking change or try again in a major release like v2 🙂

**Comment by JPustkuchen** [comment]
@bytaesu what's the final state here now in released code? This still looks to be completed?

**Comment by bytaesu** [comment]
@JPustkuchen sorry for late response

We try to avoid core schema changes and only handle them in major releases like v2 when absolutely necessary, so it's not planned for now.

**Comment by JPustkuchen** [comment]
@bytaesu  thanks for the reply! As I can see you added a tag for that. I totally understand but would be great if this could be planned for the next major release then and not get closed. Thanks!

**Comment by VIEWVIEWVIEW** [comment]
Glad to see this is now aimed to be included in 2.0.0!

Don't forget to make emails configurable as optional too 😀

**Comment by xzyfer** [comment]
Please make email optional. We have multiple services with email-less auth using phone number + sms otp

**Comment by nathan-hello** [comment]
Is there any branches or forks developing this issue? I would like to contribute.

**Comment by bytaesu** [comment]
> Hi, [@firstaxel](https://github.com/firstaxel). I'm [Dosu](https://dosu.dev), and I'm helping the better-auth team manage their backlog and am marking this issue as stale.
> 
> **Issue Summary**
> 
> * You opened this issue to request that `name` be optional during sign-up instead of required.
> * You later noted that the `name: ""` workaround no longer works because validation now requires at least one character.
> * Several commenters agreed this should be configurable or optional, especially for email/password-only flows and setups that collect first/last name separately.
> * A maintainer shared that the team discussed the change, but core schema changes are generally deferred to major releases, so this is not planned for the current release and is being considered for v2.
> 
> **Next Steps**
> 
> * If this is still relevant on the latest version of better-auth, please comment to keep the discussion open.
> * Otherwise, this issue will be automatically closed in 7 days.
> 
> Thanks for your understanding and contribution.

It's not stale
